### ISSUE TITLE
Add support for authentication database

### ISSUE THREAD

**Issue by posikifi** [qwat/pg-history-viewer#26]
Hi!

It seems to me that only Basic Authentication method of QGIS is supported. Is it possible to add also support for QGIS autentication database?

QGIS 3.18.1 on Ubuntu 20.04 with Postresql 12 and PostGIS 3.0

P

**Comment by haubourg** [comment]
+1 , BTW, this implies a refactor to not use psycopg2 but direct QGIS authentication methods. 

cc @ponceta @dsavary @lbartoletti

**Comment by ponceta** [comment]
This is true for many QGIS plugins (pum, ...), we should be able to use standard QGIS auth methods when we develop qgis python plugins.

We face also the bug that the authcfg is not supported too.

![image](https://user-images.githubusercontent.com/8614223/215472678-4763fef1-a5fb-42bf-a03a-e909fc058ebb.png)
